interfaces/browser-support: adjust base declaration for auto-connection

jdstrand opened the pull request:
During a review, pedronis discovered that the browser-support interface only
had a deny-connection constraint without a corresponding deny-auto-connection
constraint. Since constraints are evaluated independently of each other, this
meant that if a snap that used 'allow-sandbox: true' was manually approved in
the store without a snap declaration, then the interface would be
auto-connected. In practice, there were no snaps that were in the is state.
Existing snap declarations in the store for this interface have already been
adjusted to use 'allow-auto-connection: true' in addition to
'allow-connection: true'.

pedronis commented:
@jdstrand so that integration test was passing exactly because auto-connection was not blocked, this means now people testing snaps with allow-sandbox: true locally (--dangerous) need to connect manually, this is typical for most of our privileged interfaces but wasn't the case here so far.

A different approach that would keep the test as it was would be to switch to only a deny-installation: ...  rule instead of the two deny-*connection* rules.

jdstrand commented:
The testsuite changes look fine. I will adjust he browser-support documentation and notify the snap developers.

jdstrand commented:
"@jdstrand  so that integration test was passing exactly because auto-connection  was not blocked, this means now people testing snaps with allow-sandbox:  true locally (--dangerous) need to connect manually, this is typical  for most of our privileged interfaces but wasn't the case here so far."

Yesterday I adjusted the few snaps that use `allow-sandbox: true` to have `allow-auto-connection: true`. Today I updated the documentation inhttps://github.com/snapcore/snapd/wiki/Interfaces#browser-support and added a comment to https://forum.snapcraft.io/t/sandboxing-with-the-browser-support-interface/1984/4. Today I also notified the few affected snap publishers of the change in behavior via store feedback. 

Finally, I've tested that upgrades from a snapd without this PR applied to one with it applied do not disconnect the browser-support interface for an affected snap installed from the store (ie, brave from --edge). I also verified that installs of affected snaps continue to auto-connect the interface (ie, the snap declaration from yesterday is operating correctly).

jdstrand commented:
FYI, the travis test failure is unrelated and is failing due to intermittent timeouts.

jdstrand commented:
@pedronis - you added the Blocked tag. Did my subsequent comments address your concerns?

pedronis commented:
yes, I removed it already a bit ago

On Nov 17, 2017 18:54, "jdstrand" <notifications@github.com> wrote:

> @pedronis <https://github.com/pedronis> - you added the Blocked tag. Did
> my subsequent comments address your concerns?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/snapcore/snapd/pull/4238#issuecomment-345315806>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AA7jjchOwFxj1G46wHsauVridda-R1OPks5s3chHgaJpZM4QhF9U>
> .
>

jdstrand commented:
Hey, sorry. It seems github refreshes the page for some things and not others so it looked like it was still blocked.
